src/Controller/EloquaAppCloudEndpointController.php

```diff
@@ -0,0 +1,303 @@
+   * @param \Drupal\eloqua_app_cloud\Controller\RequestStack|\Symfony\Component\HttpFoundation\RequestStack $requestStack
+   * @param \Drupal\Core\Language\LanguageManagerInterface $langManager
+   * @param \Drupal\Core\Entity\EntityTypeManager $entityManager
+   * @param \Drupal\Core\Queue\QueueFactory $queue
+   * @param $plugins
+   */
+  public function __construct(ClientFactory $eloquaFactory, RequestStack $requestStack, LanguageManagerInterface $langManager, EntityTypeManager $entityManager, QueueFactory $queueFactory,array $plugins) {
+    $this->eloqua = $eloquaFactory->get();
+    $this->request = $requestStack->getCurrentRequest();
+    $this->langManager = $langManager;
+    $this->entityManager = $entityManager;
+    $this->queueFactory = $queueFactory;
+    $this->plugins = $plugins;
+  }
+
+  /**
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+
+    // Get the list of plugin managers in our "namespace".
+    foreach ($container->getServiceIds() as $serviceId) {
+      if (strpos($serviceId, 'plugin.manager.eloqua_app_cloud') === 0) {
+        $type = $container->get($serviceId);
+        $plugin_definitions = $type->getDefinitions();
+
+        foreach ($plugin_definitions as $plugin) {
+          $plugins[$plugin['id']] = $type;
+        }
+      }
+    }
+
+    return new static(
+      $container->get('eloqua.client_factory'),
+      $container->get('request_stack'),
+      $container->get('language_manager'),
+      $container->get('entity_type.manager'),
+      $container->get('queue'),
+      $plugins
+    );
+  }
+
+  /**
+   * Instantiate and store the instance ID from Eloqua.
+   *
+   * @return \Drupal\eloqua_app_cloud\Controller\JsonResponse
+   */
+  public function instantiate($eloqua_app_cloud_service) {
+    // Get the instanceID from the query parameter.
+    $instanceId = $this->request->get("instance");
+    if (empty($instanceId)) {
+      //Throw an exception, and/or return an error?
+    }
+    $pluginReferences = $this->getEntityPlugins($eloqua_app_cloud_service);
+
+    // Iterate over the ServiceEntity plugins and build a merged field list.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+      // We can only allow ONE type of API at a time.
+      if(!empty($api) && $api !== $plugin->api()){
+        // @TODO Throw an exception!
+      }
+      $api = $plugin->api();
+      // @TODO figure out how to determine field list? This pulls it from annotation, but maybe it should be in config so it can be customized?
```

Comment:
Seems like this is already determined.  The `fieldList` method is on the interface.  The base class returns the annotation data by default.  If a plugin wants to make the field list configurable, it's able to by overriding the method (though we provide no hand holding in that regard).

Reply:
Yes, I meant that as in "Do we want to continue to use annotation?"
I think that allowing config via the UI might be helpful.